source/actions3/actions/AMControlMoveAction3.cpp

```diff
@@ -151,10 +188,32 @@ void AMControlMoveAction3::onMoveFailed(int reason)
 		break;
 	}
 
-	QString fundamentalFailureMessage = QString("There was an error moving the control '%1' into position. Target: %2 %3. Actual position: %4 %5. Tolerance %6. Reason: %7.").arg(control_->name()).arg(control_->setpoint()).arg(control_->units()).arg(control_->value()).arg(control_->units()).arg(control_->tolerance()).arg(AMControl::failureExplanation(reason));
+	if(retryAvailable && attempts_ > retries_)
+		retryAvailable = false;
+
+	QString retryAvailableString = "No";
+	if(retryAvailable)
+		retryAvailableString = "Yes";
+	QString fundamentalFailureMessage = QString("There was an error moving the control '%1' into position (Retry %2, attempt %3 of %4). Target: %5 %6. Actual position: %7 %8. Tolerance %9. Reason: %10.").arg(control_->name()).arg(retryAvailableString).arg(attempts_).arg(retries_+1).arg(control_->setpoint()).arg(control_->units()).arg(control_->value()).arg(control_->units()).arg(control_->tolerance()).arg(AMControl::failureExplanation(reason));
 	// error message with reason
 	AMErrorMon::alert(this, definedFailureReason, QString("%1. Please report this problem to the beamline staff.").arg(fundamentalFailureMessage));
-	setFailed(fundamentalFailureMessage);
+
+	if(retryAvailable){
+		attempts_++;
+
+		// retry move
+		qDebug() << "Retrying move to " << destination_;
```

Comment:
Do you still want this qDebug() there?  Do you want to have an error monitor instead or is the error mon stuff handled entirely by the above error mon?